packages/config-service/src/services/globalConfig.ts

```diff
@@ -495,10 +495,11 @@ export class GlobalConfig {
       required: false,
       defaultValue: null,
     },
-    NPM_PACKAGE_VERSION: {
+    // the actual env var in the node process is npm_package_version
+    npm_package_version: {
       envName: 'npm_package_version',
       type: 'string',
-      required: false,
+      required: true,
```

Comment:
Q: does this affect the product logic or is this only required on the test side?
I see test changes for validation but it's not clear what bug was occurring and how we are confirming it won't happen again in the product side

Reply:
The bug that was occurring was the `web3_clientVersion` method was returning nothing, so I would say it affects the product.  Checking and confirming the correct product version is deployed, is one of the first steps in the release process.    The fix simply uses the new `ConfigService` and forces the version to be defined in order to start the application.  The idea here is to not have this problem again, especially when cutting a release.  There should always be a version defined in the package.json files.   A test was added to ensure a product version can be read, so that should prevent this issue from ever happening again.

Reply:
Okay. I had expect to see the place where the error was happening (web3_clientVersion in this case) have a test but I guess the claim is with coverage in the `validationService.spec.ts` there's no need anywhere else since the ConfigService would be used in that case.